### PR TITLE
libpg_query: also build shared library

### PR DESCRIPTION
**Pull request by purcell**
###### Description of changes

I'm working in an environment (Common Lisp) where I need to be able to load and interact with `libpg_query` dynamically, but the existing derivation only calls the `make` target that builds the static lib. This commit enables and installs the shared library.

Works for me on Darwin, but unsure if any further steps are idiomatic or needed, e.g. patchelf on Linux, please advise if so.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by purcell** [comment]
Also enabled tests for a little extra peace of mind.

**Comment by purcell** [comment]
Any feedback, @marsam? Also happy to add myself as a maintainer for this in future, if it would help.

File: pkgs/development/libraries/libpg_query/default.nix [hunk 1]
```diff
@@ -13,13 +13,20 @@ stdenv.mkDerivation rec {
...
-  makeFlags = [ "build" ];
+  makeFlags = [ "build" "build_shared" ];
 
   installPhase = ''
     install -Dm644 -t $out/lib libpg_query.a
     install -Dm644 -t $out/include pg_query.h
+  '' + lib.optionalString stdenv.isLinux ''
+    install -Dm644 -t $out/lib libpg_query.so
+  '' + lib.optionalString stdenv.isDarwin ''
+    install -Dm644 -t $out/lib libpg_query.dylib
```
Comment (marsam) [inline pkgs/development/libraries/libpg_query/default.nix:24]: Sorry for the delay. I think you can use:
```suggestion
    install -Dm644 -t $out/lib libpg_query${stdenv.hostPlatform.extensions.sharedLibrary}
```
Besides that, LGTM

Comment (purcell) [reply pkgs/development/libraries/libpg_query/default.nix]: Ah, thanks, I failed to find something like this the first time around! Applied.

**Review by marsam — APPROVED** [review]

**Comment by purcell** [comment]
Awesome, thanks @marsam, much appreciated

**Pull request merged**